launch_ros/launch_ros/actions/node.py

```diff
@@ -369,20 +369,15 @@ def execute(self, context: LaunchContext) -> Optional[List[Action]]:
         context.extend_locals({'ros_specific_arguments': ros_specific_arguments})
         ret = super().execute(context)
 
-        execute_process_logger = launch.logging.get_logger(self.name)
-        if '<node_name_unspecified>' in self.node_name:
-            add_node_name(context, None)
-            execute_process_logger.warning(
-                'A node without an explicit name is being launched, '
-                'this may have the unintended side effect of non-unique node names. '
-                'It is recommended to explicitly name all nodes in launch files.')
-        else:
+        if '<node_name_unspecified>' not in self.node_name:
             add_node_name(context, self.node_name)
             node_name_count = get_node_name_count(context, self.node_name)
             if node_name_count > 1:
+                execute_process_logger = launch.logging.get_logger(self.name)
                 execute_process_logger.warning(
                     'there are now {} nodes with the name {} created within this launch context'
```

Comment:
I think with the latest understanding that in order to be fully accurate we need to say "at least N nodes with the name"

Reply:
Done.